Hippo.Core/Services/EmailService.cs

```diff
@@ -34,9 +34,41 @@ public EmailService(AppDbContext dbContext, INotificationService notificationSer
             _emailSettings = emailSettings.Value;
         }
 
-        public Task<bool> AccountDecission(Account account, bool isApproved)
+        public async Task<bool> AccountDecission(Account account, bool isApproved)
         {
-            throw new NotImplementedException();
+            try
+            {
+                account = await CheckForMissingData(account);

```

Comment:
I don't really like the name or maybe the functionality of this, I can't quite decide.  One option would be to have `EnsureAccountData(account)` and then it just updates in-place.  Another would be something like `GetCompleteAccount(account)` which could return the account like it does now.  Something about `CheckFor` returning an object you need to use to modify your param doesn't quite sit well.

Reply:
Ha, I had something similar to ensure, then I changed it to what I had for harvest :)